packages/eas-cli/src/commands/build/delete.ts

```diff
@@ -0,0 +1,122 @@
+import { Build, DeleteBuildMutation, DeleteBuildMutationVariables } from '../../graphql/generated';
+import Log from '../../log';
+import { ora } from '../../ora';
+import { getDisplayNameForProjectIdAsync } from '../../project/projectUtils';
+import { confirmAsync, selectAsync } from '../../prompts';
+
+async function deleteBuildAsync(
+  graphqlClient: ExpoGraphqlClient,
+  buildId: string
+): Promise<Pick<Build, 'id'>> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<DeleteBuildMutation, DeleteBuildMutationVariables>(
+        gql`
+          mutation DeleteBuildMutation($buildId: ID!) {
+            build(buildId: $buildId) {
+              deleteBuild(buildId: $buildId) {
+                id
+              }
+            }
+          }
+        `,
+        { buildId }
+      )
+      .toPromise()
+  );
+  return data.build!.deleteBuild;
+}
+
+export async function selectBuildToDeleteAsync(
+  graphqlClient: ExpoGraphqlClient,
+  projectId: string,
+  projectDisplayName: string
+): Promise<string | null> {
+  const spinner = ora().start('Fetching builds…');
+
+  let builds;
+  try {
+    builds = await fetchBuildsAsync(graphqlClient, projectId);
+    spinner.stop();
+  } catch (error) {
+    spinner.fail(
+      `Something went wrong and we couldn't fetch the builds for the project ${projectDisplayName}.`
+    );
+    throw error;
+  }
+  if (builds.length === 0) {
+    Log.warn(`There aren't any builds for the project ${projectDisplayName}.`);
+    return null;
+  } else {
+    const buildId = await selectAsync<string>(
+      'Which build do you want to delete?',
+      builds.map(build => ({
+        title: formatBuild(build),
+        value: build.id,
+      }))
+    );
+
+    return (await confirmAsync({
+      message: 'Are you sure you want to delete it?',
+    }))
+      ? buildId
+      : null;
+  }
+}
+
+export default class BuildDelete extends EasCommand {
+  static override description = 'delete a build';
+  static override args = [{ name: 'BUILD_ID' }];
+  static override flags = {
+    ...EASNonInteractiveFlag,
+  };
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+    ...this.ContextOptions.Vcs,
+  };
+
+  async runAsync(): Promise<void> {
+    const {
+      args: { BUILD_ID: buildIdFromArg },
+      flags: { 'non-interactive': nonInteractive },
+    } = await this.parse(BuildDelete);
+    const {
+      privateProjectConfig: { projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(BuildDelete, { nonInteractive });
+    const displayName = await getDisplayNameForProjectIdAsync(graphqlClient, projectId);
+
+    if (buildIdFromArg) {
+      await ensureBuildExistsAsync(graphqlClient, buildIdFromArg);
+    }
+
+    let buildId: string | null = buildIdFromArg;
+    if (!buildId) {
+      if (nonInteractive) {
+        throw new Error('BUILD_ID must not be empty in non-interactive mode');
```

Comment:
Do users know the argument is called `BUILD_ID`? If not, maybe we should changes this to say "The build ID argument"?

Reply:
Good point

Reply:
Updated